scipy/integrate/_ivp/radau.py

```diff
@@ -327,6 +343,21 @@ def solve_lu(LU, b):
         self.solve_lu = solve_lu
         self.I = I
 
+        if not (mass is None):
+            if issparse(mass):
+                self.mass_matrix = csc_matrix(mass)
+                self.index_algebraic_vars = np.where(
```

Comment:
I could not find a sparse-specific method to find the rows that are nil. Therefore I just convert the matrix to a dense array, which is not optimal. This is however only done once, so I guess it should not affect the overall performance.

Reply:
I would rather leave this part as is, as I am not really knowledgable on sparse matrices. I chose to stick to the CSC format since it is the one used in the rest of the code, e.g. for the Jacobian. It should be easy to change that to CSR, but I'd rather keep for a separate pull request maybe.